### PR TITLE
Adding kubernetes, docker templates, TFX support and steroids on CLI

### PR DESCRIPTION
**Pull request by cardosolucas**
Some integrations and CLI commands for modern tools

### CODE REVIEW

File: python-daemon/setup.py [hunk 1]
```diff
@@ -75,6 +75,8 @@
     'idna>=2.5',
     'bleach>=1.5.0',
     'pyspark',
+    'tensorflow==2.3',
```
Comment (Wei-1) [inline python-daemon/setup.py:78]: This dependency is causing error during test: https://app.travis-ci.com/github/apache/incubator-marvin/jobs/536187758
Might need to point the `pip` to the correct artifact.

Comment (Wei-1) [reply python-daemon/setup.py:78]: Also, we also see `This package requires Rust >=1.41.0.` in the error log, and is requesting us to run `pip install --upgrade pip`.
We should simply upgrade the `pip` in the daemon and both error should be solved.

**Comment by Wei-1** [comment]
Also, I would like to know if there are separated commits so that we can separate all these dependencies bump and feature adding into several PRs.
It will be a lot easier to debug and review.

**Comment by cardosolucas** [comment]
Do you prefer to close this PR so I can commit each feature independently? I can do it this weekend.

**Comment by Wei-1** [comment]
I think it will definitely be better if we close this and open several new PRs.
But it depends on your time and resources.

**Comment by cardosolucas** [comment]
It's fine for me. You can close this one. Thanks!

**Pull request closed without merging**